### DISCUSSION TITLE
How to handle Results in the Rust/Lua boundary?

### DISCUSSION THREAD

**Discussion by nitnelave** [mlua-rs/mlua#363] [Q&A]
I'd like to have a lua function return a `Result<T, String>` (or something to that effect), and be able to understand when a function returns a `Result<T, String>`. On the Lua side, it should match the convention of returning the error as a second value, non-`nil` when there is an error.

The `std::result::Result` type implements `IntoLuaMulti` but doesn't seem to implement `FromLuaMulti`. Is that intentional or just hasn't been implemented yet?

I manage to work around that by using my own `struct MyLuaResult<T>(Result<T, String>);` but it feels like it should be provided by the crate: https://github.com/nitnelave/lldap_plugin_poc/blob/a84a9678713db21ed244f580818f090545d2449c/src/main.rs

**Comment by khvzak** [comment]
> Is that intentional or just hasn't been implemented yet?

Not intentionaly, I'll implement it!
